packages/block-editor/src/components/image-editor/use-transform-image.js

```diff
@@ -1,39 +1,73 @@
 /**
  * WordPress dependencies
  */
-import { useCallback, useMemo, useState } from '@wordpress/element';
+import { useCallback, useEffect, useMemo, useState } from '@wordpress/element';
 import { applyFilters } from '@wordpress/hooks';
+import { useImageCropper } from '@wordpress/image-cropper';
 
 export default function useTransformImage( {
 	url,
 	naturalWidth,
 	naturalHeight,
 } ) {
 	const [ editedUrl, setEditedUrl ] = useState();
-	const [ crop, setCrop ] = useState();
-	const [ position, setPosition ] = useState( { x: 0, y: 0 } );
-	const [ zoom, setZoom ] = useState( 100 );
-	const [ rotation, setRotation ] = useState( 0 );
-	const defaultAspect = naturalWidth / naturalHeight;
-	const [ aspect, setAspect ] = useState( defaultAspect );
+	const { cropperState, setCropperState } = useImageCropper();
+	const { zoom, aspectRatio, crop } = cropperState;
+
+	const setZoom = useCallback(
+		( newZoom ) => {
+			setCropperState( { zoom: newZoom } );
+		},
+		[ setCropperState ]
+	);
+
+	const setAspectRatio = useCallback(
+		( newAspect ) => {
+			setCropperState( { aspectRatio: newAspect } );
+		},
+		[ setCropperState ]
+	);
 
+	const defaultAspect = naturalWidth / naturalHeight;
+	const rotatedAspect = naturalHeight / naturalWidth;
+
+	// Initialize aspect ratio on mount or when defaultAspect changes
+	useEffect( () => {
+		setAspectRatio( defaultAspect );
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	/**
+	 * rotateClockwise rotates the image by 90° clockwise by drawing the original image onto a canvas with rotation applied,
+	 * then saves it as a new blob URL (editedUrl).
+	 * This creates a new rotated image file, bypassing the image-cropper’s CSS transform rotation.
+	 * It's a bespoke solution to ensure that the rotated image fills the entire canvas.
+	 */
+	const [ iternalRotation, setInternalRotation ] = useState( 0 );
```

Comment:
All this is to prevent regression. The current implementation bypasses the cropper package's rotation altogether by rotating a canvas blob image and reloading the cropper. 

This is presumably so the image can fill the canvas area.